ratelimit/core.py

```diff
@@ -52,32 +61,31 @@ async def __call__(self, scope: Scope, receive: Receive, send: Send) -> None:
 
         url_path = scope["path"]
         for pattern, rules in self.config.items():
-            if pattern.match(url_path):
-                # After finding the first rule that can match the path,
-                # calculate the user ID and group
-                try:
-                    user, group = await self.authenticate(scope)
-                except Exception as exc:
-                    if self.on_auth_error is not None:
-                        reponse = await self.on_auth_error(exc)
-                        return await reponse(scope, receive, send)
-                    raise exc
-
-                # Select the first rule that can be matched
-                _rules = [rule for rule in rules if group == rule.group]
-                if _rules:
-                    rule = _rules[0]
-                    break
+            if not pattern.match(url_path):
+                continue
+            # After finding the first rule that can match the path,
+            # calculate the user ID and group
+            try:
+                user, group = await self.authenticate(scope)
+            except Exception as exc:
+                if self.on_auth_error is not None:
+                    response = await self.on_auth_error(exc)
+                    return await response(scope, receive, send)
+                raise exc
+
+            # Select the first rule that can be matched
+            match_rule = list(filter(lambda r: r.group == group, rules))
+            if match_rule:
+                rule = match_rule[0]
+                break
         else:  # If no rule can match, run `self.app` and return
             return await self.app(scope, receive, send)
 
-        if not [name for name in RULENAMES if getattr(rule, name) is not None]:
+        if not list(filter(lambda name: getattr(rule, name) is not None, RULENAMES)):
```

Comment:
What is the purpose of changing the comprehension into a function call? It does not make the code faster.

Reply:
Inspired by another PR, it should be better to use `any` here.

Reply:
Ok, I will check it.